### DISCUSSION TITLE
[RFC] Data entry components support custom validation status

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#36885] [RFCs - archive]
### Summary

In order to support custom validation status, data entry components need to support two props: `validateStatus` and `hasFeedback`, which is same as the props of Form.Item. These props will still keep in `Form.Item` to avoid breaking change, which means it will not affect developers for this change.

### Basic example
```diff
<Input
++ validateStatus="error"
++ hasFeedback
/>
```

### Motivation

In the current implementation of `Form.Item`, selector with higher priority is used to override the style of data entry components in order to present validation status，which is considered highly coupled. At the mean time, data entry components do not support validation status themselves, which make it hard for developers to customize validation status in custom components. 

### Detailed

1. Data entry components support `validateStatus` and `hasFeedback` (some of the components).
a. The usage is the same as `Form.Item`.
b. Add corresponding styles to each component.
2. `validateStatus` and `hasFeedback` passed to `Form.Item` will be passed to child components through context, and will be combined with their own `validateStatus` and `hasFeedback` in props.
3. Delete the custom validation style of `Form.Item`.

Components to be updated:
- [x] Input（including Password, Search, Group)
- [x] InputNumber
- [x] DatePicker
- [x] TimePicker
- [x] Select
- [x] TreeSelect
- [x] Cascader
- [ ] ~~Radio.Button~~
- [x] Mentions
- [x] Transfer
- [x] AutoComplete

### Drawbacks

No obvious drawbacks yet.

**Comment by zombieJ** [comment]
In first phase, we can tmp not support `hasFeedback` since it more close with Form design. Expect user can use `validateStatus` for single usage only.

**Comment by afc163** [comment]
```diff
- validateStatus
+ status
```

We should drop `hasFeedback`.
